### ISSUE TITLE
What determines pet ability priority in event of an attack tie?

### ISSUE THREAD

**Issue by JorVLP** [manny405/sapai#89]
It is stated that pet priority is [determined by health in the event of a tie](https://github.com/manny405/sapai/blob/ec22d8306ac4f50f1f3c7df1b5a5ec85b367954d/sapai/battle.py#L225).

In a versus battle against myself, running these mosquitos over and over always lead to the right-hand mosquito going first
- When they had equal 2-2 stats at level 1
- When left was 4-4 level 2, and right was 4-4 level 1
- When left was 4-5 level 2, and right was 4-4 level 1

Not sure if this is versus exclusive, or effects arena mode the same, but it seems that **health does not prioritise pets** at all. 
Also seems that on attack ties, the **outcome is not random** as something (maybe player data?) always favours one team over the other.
![Screenshot_4](https://user-images.githubusercontent.com/47005135/192169552-ae4be01b-85cd-4274-b0f9-d46c1c85ee81.png)

Should the health tie-break be removed? (and possibly the random tie-break)?

**Comment by manny405** [comment]
You are correct. This was done very early and is obviously not how the game operates. 

The health tie-break should be removed and a random tie-break needs to be added. With this change, random-state updating will need to be included so battles can continue to be seeded for a deterministic outcome. 

For example: https://github.com/manny405/sapai/blob/8709b8f40ee939c0eb86865b0bcf5328adc6f231/tests/test_effect_queue.py#L226-L246

**Comment by proudno0b** [comment]
I'm pretty sure that pet ability is determined by attack amount i.e a 12 attack dodo will buff an 11 attack leopard before the leopard uses its snipe ability, but it's unclear what would happen in the result of a tie
